### PR TITLE
Blog on configuration automation for Istio multicluster with local control planes

### PR DESCRIPTION
**Pull request by aattuluri**

### CODE REVIEW

File: content/en/blog/2019/multi-cluster-mesh-automation/index.md [hunk 1]
```diff
@@ -0,0 +1,130 @@
...
+title: Multi-mesh configuration and service discovery using Admiral
+subtitle: Configuration automation for Istio multi-mesh.
+description: Automating Istio configuration for Istio deployments (clusters) that work as a single mesh.
+publishdate: 2019-12-28
+attribution: Anil Attuluri (Intuit), Jason Webb (Intuit)
+keywords: [traffic-management,automation,configuration,multicluster,multi-mesh,gateway,federated,globalidentifer]
+target_release: 1.5
+---
+
+When we read the blog post [Multi-Mesh Deployments for Isolation and Boundary Protection](https://istio.io/blog/2019/isolated-clusters/) we could immediately relate to some of the problems mentioned and how we are using [Admiral](https://github.com/istio-ecosystem/admiral), an open source project under istio-ecosystem to solve these.
```
Comment (benchi) [inline content/en/blog/2019/multi-cluster-mesh-automation/index.md]: mentioned and... -> mentioned, and we thought we could share how we are using ... to solve similar issues.

File: content/en/blog/2019/multi-cluster-mesh-automation/index.md [hunk 2]
```diff
@@ -0,0 +1,130 @@
...
+
+## Background
+
+We were off to a good start with Istio but quickly realized the configuration for multi-cluster was complicated and would be challenging to maintain over time. We picked an opinionated version of [Multi-Cluster Istio Service Mesh with local control planes](https://istio.io/docs/setup/install/multicluster/gateways/#deploy-the-istio-control-plane-in-each-cluster) for scalability and other operational reasons. Having picked this model, we had the following  key requirements to solve for before allowing wide adoption of Istio Service Mesh:
```
Comment (benchi) [inline content/en/blog/2019/multi-cluster-mesh-automation/index.md]: Having picked -> Having selected this model

File: content/en/blog/2019/multi-cluster-mesh-automation/index.md [hunk 3]
```diff
@@ -0,0 +1,130 @@
...
+- Creation of service DNS decoupled from the namespace (Uniform service naming mentioned [here](https://istio.io/blog/2019/isolated-clusters/#features-of-multi-mesh-deployments))
+- Service Discovery across many clusters
+- Support Active-Active & HA/DR deployments with services being deployed in globally unique namespaces in clusters in different regions.
```
Comment (benchi) [inline content/en/blog/2019/multi-cluster-mesh-automation/index.md]: Active-Active Support with HA/DR deployments. We also needed to support these critical resiliency patterns with services being deployed in globally unique namespaces across discrete clusters.

File: content/en/blog/2019/multi-cluster-mesh-automation/index.md [hunk 4]
```diff
@@ -0,0 +1,130 @@
...
+
+We have over 160 k8s clusters and the namespace naming schema utilized created a globally unique name across these clusters for every namespace. This meant the same service binary deployed in different regions would be run in namespaces with different names. This would mean that `foo.namespace.global` (routing strategy mentioned in [Multicluster version routing](https://istio.io/blog/2019/multicluster-version-routing/) name would no longer work across clusters. We needed a globally unique and discoverable service DNS that resolves service instances in multiple clusters, each instance running/addressable with its own unique k8s FQDN. (Example: `foo.global` would resolve to both `foo.uswest2.svc.cluster.local` & `foo.useast2.svc.cluster.local` if foo is running in two k8s clusters with unique namespaces)
```
Comment (benchi) [inline content/en/blog/2019/multi-cluster-mesh-automation/index.md]: and the namespace ... for every namespace -> , and we use a globally unique namespace name across all clusters. 

"service binary" -> "service workload"

File: content/en/blog/2019/multi-cluster-mesh-automation/index.md [hunk 5]
```diff
@@ -0,0 +1,130 @@
...
+Also, in our case, services needed additional DNS names with different resolution and global routing properties. For example, `default.payments.global` would resolve locally first, and then route to a remote instance (using topology routing) while `default.payments-west.global` and `default.payments-east.global` would always resolve to the respective regions (such names were needed for testing).
+
+## Contextual Configuration
+As we investigated how to solve the aforementioned issues, it became apparent that configuration needed to be contextual. By contextual, the configuration would be different depending on where the services and their clients are deployed.
```
Comment (benchi) [inline content/en/blog/2019/multi-cluster-mesh-automation/index.md]: By contextual, the configuration would be different depending on where the services and their clients are deployed. 
->
In other words, each cluster would need to be delivered a configuration that was specifically tailored for its view of the world.

File: content/en/blog/2019/multi-cluster-mesh-automation/index.md [hunk 6]
```diff
@@ -0,0 +1,130 @@
...
+
+Here is an example:
+We have a payments service consumed by orders and reports. The payments service has a HA/DR deployment across us-east (cluster 3) and us-west (cluster 2). Payments service is deployed in namespaces with different names in each region. The orders service is deployed in different cluster as payments in us-west (cluster 1). The reports service is deployed in the same cluster as payments in us-west (cluster 2).
```
Comment (benchi) [inline content/en/blog/2019/multi-cluster-mesh-automation/index.md]: in different cluster -> in a different cluster

**Comment by aattuluri** [comment]
@adammil2000 
I think the PR is ready for review, I do not have access to add google drawings to [team drive](https://drive.google.com/drive/u/0/folders/1jczscJueUBR3IOvH30q9HAZX0jC7GSyW). I can add the google drawings for the pictures in the blog if I can get access.

**Comment by aattuluri** [comment]
@googlebot I consent

**Comment by adammil2000** [comment]
> @adammil2000
> I think the PR is ready for review, I do not have access to add google drawings to [team drive](https://drive.google.com/drive/u/0/folders/1jczscJueUBR3IOvH30q9HAZX0jC7GSyW). I can add the google drawings for the pictures in the blog if I can get access.

@frankbu Do you have permissions to let Anil put the art sources into the team drive? Thanks!

**Comment by frankbu** [comment]
@aattuluri There are a few formatting and minor grammatical errors in this doc, and also a bit of confusion about multi-cluster vs multi-mesh configuration. Unfortunately this PR was merged prematurely, so I created a followup PR with my suggested changes. PTAL at https://github.com/istio/istio.io/pull/6286

**Pull request merged**